### PR TITLE
Merged Mining - Tari Conduit Application

### PR DESCRIPTION
**Pull request by SWvheerden**
## Description

Added application which XMRig can be connected to which forwards requests back and forth between XMRig and MoneroD.
Added parsing of Monero requests from XMRig and adding of merge mining data to response from MoneroD.
Added integration to Tari Base Node for merged mining.
Added ReadMe.
Added GRPC refector

Code by @StriderDM  and @SWvheerden

## Types of changes

* [ ] Bug fix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:

* [ ] I'm merging against the `development` branch.
* [ ] I ran `cargo-fmt --all` before pushing.
* [ ] I have squashed my commits into a single commit.
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

### CODE REVIEW

File: applications/tari_app_grpc/proto/wallet.proto [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+import "types.proto";
+import "google/protobuf/wrappers.proto";
+import "google/protobuf/timestamp.proto";
+
+
+// The gRPC interface for interacting with the wallet.
+service Wallet {
+    // This returns a coinbase transction
+    rpc GetCoinbase(CoinbaseRequest) returns (stream CoinbaseSupply);
```
Comment (stringhandler) [inline applications/tari_app_grpc/proto/wallet.proto]: The choice to use a stream here seems weird. Won't it be a single result?

**Review by stringhandler — DISMISSED** [review]
I ran this locally, but it's difficult to test it fully. I think there may be some breaking changes on the GRPC interface for the explorer etc, but I think we can merge it and then fix those. Will test it for backwards compatibility, but otherwise I'm happy to merge it

File: applications/tari_app_grpc/proto/wallet.proto (hunk 1 above)
Comment (SWvheerden) [reply applications/tari_app_grpc/proto/wallet.proto]: haha good catch. It should not be a stream. This is what you get if you write the code without implementing it yet.

**Comment by stringhandler** [comment]
Tested this locally, and synced fine with the current chain

**Pull request merged**